src/standard/effectBuilder.html

```diff
@@ -1,214 +1,269 @@
+        this._addPropertyEffect(arg.model, 'complexObserver', {
+          method: sig.method,
+          args: sig.args,
+          trigger: arg
+        });
+      }, this);
+    },
+
+    _addAnnotationEffects: function(notes) {
+      // create a virtual annotation list, must be concretized at instance time
+      this._nodes = [];
+      // process annotations that have been parsed from template
+      notes.forEach(function(note) {
+        // where to find the node in the concretized list
+        var index = this._nodes.push(note) - 1;
+        note.bindings.forEach(function(binding) {
+          this._addAnnotationEffect(binding, index);
+        }, this);
+      }, this);
+    },
+
+    _addAnnotationEffect: function(note, index) {
+      // TODO(sjmiles): annotations have 'effects' proper and 'listener'
+      if (Polymer.Bind._shouldAddListener(note)) {
+        // <node>.on.<dash-case-property>-changed: <path> = e.detail.value
+        Polymer.Bind._addAnnotatedListener(this, index,
+          note.name, note.value, note.event);
+      }
+      if (note.signature) {
+        this._addAnnotatedComputationEffect(note, index);
+      } else {
+        // capture the node index
+        note.index = index;
+        // add 'annotation' binding effect for property 'model'
+        this._addPropertyEffect(note.model, 'annotation', note);
+      }
+    },
+
+    _addAnnotatedComputationEffect: function(note, index) {
+      var sig = note.signature;
+      if (sig.static) {
+        this.__addAnnotatedComputationEffect('__static__', index, note, sig, null);
+      } else {
+        sig.args.forEach(function(arg) {
+          if (!arg.literal) {
+            this.__addAnnotatedComputationEffect(arg.model, index, note, sig, arg);
+          }
+        }, this);
+      }
+    },
+
+    __addAnnotatedComputationEffect: function(property, index, note, sig, trigger) {
+      this._addPropertyEffect(property, 'annotatedComputation', {
+        index: index,
+        kind: note.kind,
+        property: note.name,
+        negate: note.negate,
+        method: sig.method,
+        args: sig.args,
+        trigger: trigger
+      });
+    },
+
+    // method expressions are of the form: `name([arg1, arg2, .... argn])`
+    _parseMethod: function(expression) {
+      var m = expression.match(/(\w*)\((.*)\)/);
+      if (m) {
+        // replace escaped commas with comma entity, split on un-escaped commas
+        var args = m[2].replace(/\\,/g, '&comma;').split(',');
+        return this._parseArgs(args, { method: m[1], static: true });
+      }
+    },
+
+    _parseArgs: function(argList, effect) {
+      effect.args = argList.map(function(rawArg) {
+        var arg = this._parseArg(rawArg);
+        if (!arg.literal) {
+          effect.static = false;
+        }
+        return arg;
+      }, this);
+      return effect;
+    },
+
+    _parseArg: function(rawArg) {
+      // clean up whitespace
+      var arg = String(rawArg).trim()
```

Comment:
Seems like `rawArg` is guaranteed to be a string (comes from `split`)...  seems safe to just do `rawArg.trim(). ...`